src/components/ui/hook-form/input-phone/index.tsx

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useState } from 'react';
+import {
+    PhoneInput,
+    PhoneInputProps,
+    defaultCountries,
+    parseCountry,
+} from 'react-international-phone';
+import 'react-international-phone/style.css';
+import styled from '@emotion/styled';
+import { Flex } from 'antd';
+import {
+    UseControllerProps,
+    useController,
+    useFormContext,
+} from 'react-hook-form';
+import { AnimatePresence } from 'framer-motion';
+import cn from 'classnames';
+import { COLORS } from '~variables';
+import { ErrorMessage, Label } from '../common';
+
+type InputPhoneProps = {
+    className?: string;
+    label?: string;
+    listCountries?: string[];
+} & PhoneInputProps &
+    UseControllerProps;
+
+const InputPhoneBase = ({
+    name,
+    label,
+    rules,
+    defaultValue = '',
+    className,
+    ...props
+}: InputPhoneProps) => {
+    const { control, trigger } = useFormContext();
+    const { field, fieldState } = useController({
+        name,
+        defaultValue,
+        control,
+        rules,
+    });
+    const error = fieldState.error ? fieldState.error.message : '';
+
+    const countries = defaultCountries.filter((country) => {
```

Comment:
Not you calculate it for every rerender right? It's pretty much like the constant, so we can calculate it globally in this file. Just once